homeassistant/components/openweathermap/weather_update_coordinator.py

```diff
@@ -189,6 +193,13 @@ def _calc_precipitation(rain, snow):
     @staticmethod
     def _get_condition(weather_code):
         """Get weather condition from weather data."""
+        if weather_code == WEATHER_CODE_SUNNY_OR_CLEAR_NIGHT:
+            hour = dt.now().hour
+
+            if 6 <= hour <= 18:
```

Comment:
Large differences in sun-up and sun-down times during the seasons.
Could this somehow be connected to the local sun-up and down-times? Via the HA Sun integration?

Reply:
Or even better, use suntime: https://pypi.org/project/suntime/